src/modules/sync/lib/ipc-handlers.ts

```diff
@@ -158,16 +166,94 @@ export async function pushArchive(
 		throw new Error( 'No token found' );
 	}
 
+	let isUploadingPaused = false;
+	const file = fs.createReadStream( archivePath );
+	const fileSize = fs.statSync( archivePath ).size;
+	const filename = path.basename( archivePath );
+
+	const cleanup = () => {
+		file.destroy();
+		file.close();
+	};
+
+	const attachmentId = await new Promise< string >( ( resolve, reject ) => {
+		const upload = new Upload( file, {
+			endpoint: `${ STUDIO_FILE_UPLOADS_ENDPOINT_BASE }/${ remoteSiteId }`,
+			chunkSize: STUDIO_FILE_UPLOADS_CHUNK_SIZE,
+			retryDelays: STUDIO_FILE_UPLOADS_RETRY_DELAYS,
+			overridePatchMethod: true,
+			removeFingerprintOnSuccess: true,
+			storeFingerprintForResuming: true,
+			headers: {
+				Authorization: `Bearer ${ token.accessToken }`,
+			},
+			metadata: {
+				filename,
+				filetype: 'application/gzip',
+			},
+			uploadSize: fileSize,
+			onBeforeRequest: ( req: HttpRequest ) => {
+				if ( req.getMethod() === 'HEAD' ) {
+					// @ts-expect-error We need to override the method to get the response headers.
+					req._method = 'GET';
+					req.setHeader( 'X-HTTP-Method-Override', 'HEAD' );
+				}
+			},
+			onError: ( error ) => {
+				cleanup();
+				console.error( '[TUS] Upload error', error );
+				reject( error );
+			},
+			onProgress: ( bytesSent, bytesTotal ) => {
+				if ( isUploadingPaused ) {
+					isUploadingPaused = false;
+					void sendIpcEventToRenderer( 'studio-file-upload-resumed', {
+						selectedSiteId: selectedSiteId,
+						remoteSiteId: remoteSiteId,
+					} );
+					console.log( '[TUS] Upload resumed' );
+				}
+
+				if ( bytesTotal > 0 ) {
+					const progress = ( bytesSent / bytesTotal ) * 100;
+					console.log( '[TUS] Upload progress: %s%%', progress.toFixed( 2 ) );
+				}
+			},
+			onSuccess: ( payload: { lastResponse: HttpResponse } ) => {
```

Comment:
```suggestion
			onSuccess: ( payload ) => {
```
`tus-js-client` already provides the type.

Reply:
Updated in [c1efbe1](https://github.com/Automattic/studio/pull/2212/commits/c1efbe13c0b111cb0f49e74ed76bc9f3eaffccf2).